frontend/components/color_encoding_to_sound/BasePage.js

```diff
@@ -2,63 +2,123 @@ import React from "react";
 import {SketchPicker} from 'react-color';
 import {fetchPost} from "../../common";
 import SliderInstrument from "../instruments/SliderInstrument";
+import PaletteColor from "../color_encoding_to_sound/PaletteColor";
 
 class BasePage extends React.Component {
-    state = {
-        color: {
-            r: 51,
-            g: 51,
-            b: 51,
-        },
-        result: [],
-        listOfColors: [],
-    };
+    constructor(props) {
+        super(props);
+
+        this.handlePaletteClick = this.handlePaletteClick.bind(this);
+        this.state = {
+            result: [],
+            selected: 0,
+            instrumentGenerated: false,
+            colorPickerColor: {
+                    r: 51,
+                    g: 51,
+                    b: 51,
+            },
+            listOfColors: [
+                {
+                    r: 0,
+                    g: 0,
+                    b: 0,
+                },
+                {
+                    r: 0,
+                    g: 0,
+                    b: 0,
+                },
+                {
+                    r: 0,
+                    g: 0,
+                    b: 0,
+                },
+                {
+                    r: 0,
+                    g: 0,
+                    b: 0,
+                },
+                {
+                    r: 0,
+                    g: 0,
+                    b: 0,
+                },
+                {
+                    r: 0,
+                    g: 0,
+                    b: 0,
+                },
+                {
+                    r: 0,
+                    g: 0,
+                    b: 0,
+                }
+            ],
+        };
+    }
+
+    handlePaletteClick = (e) => {
+        this.setState({ selected: Number(e.target.id) });
+    }
 
     handleChangeComplete = (color) => {
-        this.setState({color: color.rgb});
+        const currentColorState = [ ...this.state.listOfColors ];
+        currentColorState[this.state.selected] = color.rgb;
+        this.setState({ listOfColors: [ ...currentColorState ], colorPickerColor: color.rgb });
     };
 
     handleSubmit = () => {
-        const requestBody = {color: this.state.color};
+        const requestBody = {listOfColors: this.state.listOfColors};
         const responseCallbackFunc = responseDict => {
-            const tempColor = this.state.color;
             this.setState({
                 result: responseDict,
-                listOfColors: this.state.listOfColors.concat(tempColor),
+                instrumentGenerated: true
             });
-            console.log(this.state.listOfColors);
+            alert("Instrument has been updated!");
```

Comment:
This alert feels a little funky in the UI: maybe we can come up with a better signal in the interface to let the user know that the instrument has been updated.